apps/designer/app/designer/features/topbar/share.tsx

```diff
@@ -1,75 +1,126 @@
-  const [isOpen, setIsOpen] = useIsShareDialogOpen();
-  return (
-    <DeprecatedPopover open={isOpen} onOpenChange={setIsOpen}>
-      <DeprecatedPopoverTrigger asChild aria-label="Share project">
-        <DeprecatedIconButton>
-          <Share1Icon />
-        </DeprecatedIconButton>
-      </DeprecatedPopoverTrigger>
-      <DeprecatedPopoverPortal>
-        <Content {...props} />
-      </DeprecatedPopoverPortal>
-    </DeprecatedPopover>
-  );
-};
+const useShareProjectContainer = () => {
+  const { data: links, load } = trpc.findMany.useQuery();
+  const { send: createToken } = trpc.create.useMutation();
+  const { send: removeToken } = trpc.remove.useMutation();
+  const { send: updateToken } = trpc.update.useMutation();
+
+  const [project] = useProject();
+  const projectId = project?.id;
+
+  useEffect(() => {
+    if (projectId === undefined) {
+      return;
+    }
+
+    load({ projectId });
+  }, [load, projectId]);
+
+  const handleChangeDebounced = useDebouncedCallback((link: LinkOptions) => {
+    if (projectId === undefined) {
+      return;
+    }
+    updateToken({
+      projectId: projectId,
+      token: link.token,
+      relation: link.relation,
+      name: link.name,
+    });
+  }, 100);
 
-export const ShareButton = () => {
-  const handleChange = (link: LinkOptions) => {
-    // @todo implement
-  };
   const handleDelete = (link: LinkOptions) => {
-    // @todo implement
+    if (projectId === undefined) {
+      return;
+    }
+
+    removeToken({ projectId: projectId, token: link.token });
   };
+
   const handleCreate = () => {
-    // @todo implement
+    if (projectId === undefined) {
+      return;
+    }
+
+    createToken({
+      projectId: projectId,
+      relation: "viewers",
+      name: "Custom link",
+    });
+  };
+
+  return {
+    links,
+    handleChangeDebounced,
+    handleDelete,
+    handleCreate,
   };
+};
+
+/**
+ * we place the logic inside Popover so that the fetcher does not exist outside of it.
+ * Then remix will not call `trpc.findMany.useQuery` if Popover is closed
+ */
+const ShareProjectContainer = ({ designerUrl }: ShareButtonProps) => {
+  const {
+    links,
+    // handleChange,
+    handleChangeDebounced,
+    handleDelete,
+    handleCreate,
+  } = useShareProjectContainer();
+
   return (
     <ShareProject
-      onChange={handleChange}
+      links={links}
+      onChange={handleChangeDebounced}
       onDelete={handleDelete}
       onCreate={handleCreate}
-    >
-      <Button>Share</Button>
-    </ShareProject>
+      designerUrl={designerUrl}
+    />
+  );
+};
+
+export const ShareButton = ({ designerUrl }: ShareButtonProps) => {
+  const classZIndex = css({ zIndex: theme.zIndices[1] });
+
+  return (
+    <FloatingPanelPopover modal>
+      <FloatingPanelPopoverTrigger asChild>
+        <Button>Share</Button>
+      </FloatingPanelPopoverTrigger>
+
+      <FloatingPanelPopoverContent className={classZIndex()}>
```

Comment:
```suggestion
      <FloatingPanelPopoverContent css={{ zIndex: theme.zIndices[1] }}>
```

Reply:
FloatingPanelPopoverContent now doesn't support css prop. Internals need to be wrapped with styled etc.